module/zfs/dmu_recv.c

```diff
@@ -1716,52 +1814,30 @@ receive_write(struct receive_writer_arg *rwa, struct drr_write *drrw,
 	    drrw->drr_offset < rwa->last_offset)) {
 		return (SET_ERROR(EINVAL));
 	}
+
+	struct receive_record_arg *first_rrd = list_head(&rwa->write_batch);
+	if (first_rrd != NULL && (drrw->drr_object !=
+	    first_rrd->header.drr_u.drr_write.drr_object ||
+	    drrw->drr_offset >=
+	    first_rrd->header.drr_u.drr_write.drr_offset +
+	    zfs_recv_write_batch_size)) {
```

Comment:
I'm concerned that we're not setting an upper bound on the `zfs_recv_write_batch_size`.  Why don't we cap this at half of maximum transaction size, and update the description accordingly (`DMU_MAX_ACCESS / 2`).  That's 32M which should be plenty large.

Reply:
sure thing.